src/InteractiveWindow/Editor/InteractiveWindow.UIThreadOnly.cs

```diff
@@ -2803,7 +3137,18 @@ private void AddStandardInputSpan()
             /// <summary>Implements <see cref="IInteractiveWindowOperations.BreakLine"/>.</summary>
             public bool BreakLine()
             {
-                return HandlePostServicesReturn(false);
+                using (var transaction = UndoHistory_TestOnly.CreateTransaction(InteractiveWindowResources.BreakLine))
```

Comment:
Would it be possible to just not create the transaction when reading standard input?  `CreateTransaction` could return null and the `Complete` call could use `?.`.